core/lib/spree/testing_support/common_rake.rb

```diff
@@ -24,7 +24,7 @@
     begin
       require "generators/#{ENV['LIB_NAME']}/install/install_generator"
       puts 'Running extension installation generator...'
-      "#{ENV['LIB_NAMESPACE'] || ENV['LIB_NAME'].camelize}::Generators::InstallGenerator".constantize.start(["--auto-run-migrations"])
+      sh "bin/rails generate #{ENV['LIB_NAMESPACE']&.underscore || ENV['LIB_NAME']}:install --auto-run-migrations"
```

Comment:
Hi @mdesantis! I saw that you introduced `LIB_NAMESPACE` in https://github.com/solidusio/solidus/pull/3161. Am I correct in assuming that its value can be underscored and prefixed in the `<extension>:install` generator call? For example, with `LIB_NAMESPACE=SolidusGraphQL`, the resulting call would be:

```sh
bin/rails generate solidus_graph_ql:install --auto-run-migrations
```

I haven't had time to test this today, but I'll test this tomorrow.

Reply:
@mdesantis I have just noticed now that you're no longer working with Nebulab. You can ignore my question earlier :)

@waiting-for-dev @kennyadsl Unfortunately, I wasn't able to get `ENV['LIB_NAMESPACE']` to work with this change. I tested `ENV['LIB_NAMESPACE']` against my SolidusHelloWorld extension by renaming the generator's module from `SolidusHelloWorld::Generators::InstallGenerator` to `UpdatedSolidusHelloWorld::Generators::InstallGenerator`:

```diff
solidus_hello_world 17:41:51 $ git diff HEAD
diff --git a/lib/generators/solidus_hello_world/install/install_generator.rb b/lib/generators/solidus_hello_world/install/install_generator.rb
index 65d613c..0802fe5 100644
--- a/lib/generators/solidus_hello_world/install/install_generator.rb
+++ b/lib/generators/solidus_hello_world/install/install_generator.rb
@@ -1,6 +1,6 @@
 # frozen_string_literal: true

-module SolidusHelloWorld
+module UpdatedSolidusHelloWorld
   module Generators
     class InstallGenerator < Rails::Generators::Base
       class_option :auto_run_migrations, type: :boolean, default: false
```

I then tried to create the test app as follows:

```sh
LIB_NAMESPACE=UpdatedSolidusHelloWorld bundle exec rake extension:test_app
```

However, Rails is not able to find the generator:

```
solidus_hello_world 17:38:39 $ LIB_NAMESPACE=UpdatedSolidusHelloWorld bundle exec rake extension:test_app
Generating dummy Rails application...
      create  config/master.key
Copied migration 20220317093906_create_active_storage_tables.active_storage.rb from active_storage
Created database 'db/solidus_test.sqlite3'
Setting up dummy database...
bin/rails db:environment:set RAILS_ENV=test
bin/rails db:drop db:create db:migrate VERBOSE=false RAILS_ENV=test
Running extension installation generator...
bin/rails generate updated_solidus_hello_world:install --auto-run-migrations
Could not find generator 'updated_solidus_hello_world:install'.
Run `bin/rails generate --help` for more options.
cd /home/gsmendoza/repos/gsmendoza/solidus_hello_world
```

Is there any reason to keep the `LIB_NAMESPACE` environment variable? The variable was introduced in https://github.com/solidusio/solidus/pull/3161 to allow the `test_app` task to work with the `SolidusGraphQL` module name. However, the GraphQL extension is already using the more Inflector-friendly module name `SolidusGraphqlApi` (https://github.com/solidusio-contrib/solidus_graphql_api). 

The following searches in the solidusio and solidusio-contrib orgs also do not show any Code results for `LIB_NAMESPACE`:

* https://github.com/search?q=org%3Asolidusio+LIB_NAMESPACE
* https://github.com/search?q=org%3Asolidusio-contrib+LIB_NAMESPACE

Reply:
If unused, we can definitely remove `LIB_NAMESPACE` from here.

Reply:
Hey @gsmendoza, I wonder if that didn't work because you didn't rename the directory `solidus_hello_world` to `updated_solidus_hello_world`. If that works, I think it can be helpful to leave a way out from the default inferred name.

Reply:
Hi @waiting-for-dev! I've confirmed that if we make this additional change to `common:test_app`, we can continue to use `LIB_NAMESPACE`:

```rb
# core/lib/spree/testing_support/common_rake.rb
# ...
    begin
      # Add `ENV['LIB_NAMESPACE']&.underscore` in the line below:
      require "generators/#{ENV['LIB_NAMESPACE']&.underscore || ENV['LIB_NAME']}/install/install_generator" 

      puts 'Running extension installation generator...'
      sh "bin/rails generate #{ENV['LIB_NAMESPACE']&.underscore || ENV['LIB_NAME']}:install --auto-run-migrations"
    rescue LoadError
      # No extension generator to run
    end
# ...
```

However, in the SolidusHelloWorld extension, we'll need to make the following corresponding changes:

1. Rename the module for the generator, e.g. from `SolidusHelloWorld::Generators::InstallGenerator` to `UpdatedSolidusHelloWorld::Generators::InstallGenerator`.
2. Rename the generator directory e.g. from `lib/generators/solidus_hello_world` to `lib/generators/updated_solidus_hello_world`.

I intend to continue with the update to the require statement, since it provides a way to continue using the `LIB_NAMESPACE` variable.

Let me know if you have additional thoughts!

Reply:
Great, @gsmendoza! Thanks for looking into it.